src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,284 @@
+                                                                pal_high_frequency,
+                                                                device_a['userId'])
+        ppas.append(ppa)
+        pals.append(pal)
+        # ESC senosrs
+        esc_sensors = []
+        esc_sensors.append(json.load(
+              open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json'))))
+        config = {
+            'registrationRequests': devices,
+            'conditionalRegistrationData': conditionals,
+            'grants': grants,
+            'ppas': ppas,
+            'pals': pals,
+            'esc_sensors' : esc_sensors
+        }
+        writeConfig(filename, config)
+    def assertEqualToDeviceParamOrIfExistWithDefault(self, attr_name, parent_attr_in_record,\
+                                          parent_attr_in_device, attr_defaule_value):
+        if attr_name in parent_attr_in_device:
+           self.assertEqual(parent_attr_in_device[attr_name],\
+             parent_attr_in_record[attr_name])
+        elif attr_name in parent_attr_in_record:
+            self.assertEqual(parent_attr_in_record[attr_name], attr_defaule_value)
+        
+    def assertCbsdRecord(self, registration_request, grant_request, grant_response, cbsd_dump_data):
+        for index, device in enumerate(registration_request):
+            record_id = 'cbsd/'+ device['fccId']+'/'+ hashlib.sha1(device['cbsdSerialNumber']).hexdigest()
+            cbsd_record = [record['registration'] for record in cbsd_dump_data if record['id'] == record_id]
+            self.assertEqual(1, len(cbsd_record))
+            # required parameters
+            self.assertEqual(device['cbsdCategory'], cbsd_record[0]['cbsdCategory'])
+            self.assertEqual(device['fccId'], cbsd_record[0]['fccId'])
+            self.assertEqual(device['airInterface']['radioTechnology'],\
+                             cbsd_record[0]['airInterface']['radioTechnology'])
+            self.assertEqual(device['installationParam']['latitude'],\
+                             cbsd_record[0]['installationParam']['latitude'])
+            self.assertEqual(device['installationParam']['longitude'],\
+                             cbsd_record[0]['installationParam']['longitude'])
+            self.assertEqual(device['installationParam']['height'],\
+                             cbsd_record[0]['installationParam']['height'])
+            self.assertEqual(device['installationParam']['heightType'], \
+                             cbsd_record[0]['installationParam']['heightType'])
+            self.assertEqual(device['installationParam']['antennaGain'], \
+                             cbsd_record[0]['installationParam']['antennaGain'])
+             # parameters should exist in record if exist in device,\
+             # if exist only in record, its value should be equal to a default value          
+            self.assertEqualToDeviceParamOrIfExistWithDefault('indoorDeployment', \
```

Comment:
I think that all REG-Conditional fields are required to be in the SAS-SAS exchange (whether they came from CPI signed data or pre-loaded conditional reg data)... so we can require this to exist, right?

(This doesn't apply to e.g. antennaAzimuth though because that one is REG-Conditional for Cat B and Optional for Cat A... so I guess we need more complex logic there :/ )